### PR TITLE
Remove clang-pseudo

### PR DESCRIPTION
**Pull request by AaronBallman**
The functionality is incomplete and the authors have since shifted gears to other work, so this is effectively unmaintained.

The original design document for clang-pseudo can be found at:
https://docs.google.com/document/d/1eGkTOsFja63wsv8v0vd5JdoTonj-NlN3ujGF0T7xDbM/edit
in case anyone wishes to pick this project back up again in the future.

Original RFC: https://discourse.llvm.org/t/removing-pseudo-parser/71131/

### CODE REVIEW

File: clang-tools-extra/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -43,6 +43,8 @@ infrastructure are described first, followed by tool-specific sections.
 Major New Features
 ------------------
 
+- The ``clang-pseudo`` tool has been removed.
```
Comment (erichkeane) [inline clang-tools-extra/docs/ReleaseNotes.rst:46]: A link to the RFC might be nice here, lets folks see the WHY.

**Review by erichkeane — APPROVED** [review]
I think the release note should be mroe detailed and give some help as to the WHY here, and not just 'we removed it'.

File: clang-tools-extra/clangd/SemanticSelection.cpp [hunk 2]
```diff
@@ -181,16 +181,16 @@ llvm::Expected<std::vector<FoldingRange>> getFoldingRanges(ParsedAST &AST) {
 // Related issue: https://github.com/clangd/clangd/issues/310
 llvm::Expected<std::vector<FoldingRange>>
 getFoldingRanges(const std::string &Code, bool LineFoldingOnly) {
```
Comment (nico) [inline clang-tools-extra/clangd/SemanticSelection.cpp:183]: Should we revert 70914aa631561aa9a29681bfe5159b4ea6952060 instead?

**Comment by cor3ntin** [comment]
Thanks for doing that Aaron.
LGTM on the clang side of things (and it does what the rfc proposed) - but I'll let a clangd maintainer approve.

I was really excited by the pseudo parser as I think it would have been a very useful tool with many use cases.
However, since the RFC was made, no one has stepped up to take on maitainership (and that would be a big commitment).

So I think ripping it out is the right thing to do.

**Comment by zyn0217** [comment]
For visibility, see the last attempt at removal: https://github.com/llvm/llvm-project/pull/80081

AFAIK, some Clangd features e.g. code folding, are still relying on it, so ditching it completely is probably not an option at the moment.

**Comment by kadircet** [comment]
thanks a lot for taking care of this @AaronBallman, I think the only concern here is not regressing clangd functionality. We can figure out how we should trim down the bits that're moving into clangd later if need be. I think this LG, but cc @hokein  as well.

File: clang-tools-extra/clangd/SemanticSelection.cpp (hunk 2 above)
Comment (hokein) [reply clang-tools-extra/clangd/SemanticSelection.cpp:183]: I think we should keep it in this patch to avoid changing the current behavior of clangd.

**Review by hokein — APPROVED** [review]
This looks good.

**Pull request merged**